backend/fms_core/viewsets/_fetch_data.py

```diff
@@ -2,57 +2,88 @@
 import json
 from typing import List
 from collections import defaultdict
-from django.db.models import OuterRef, Subquery, Count, Prefetch, Q, ExpressionWrapper, BooleanField, Exists
+from django.db.models import Q, ExpressionWrapper, BooleanField
 
 from fms.settings import REST_FRAMEWORK
-from fms_core.models import Sample, DerivedBySample, DerivedSample, SampleLineage, ProcessMeasurement, Project
+from fms_core.models import Sample, DerivedSample, SampleLineage, ProcessMeasurement
 from fms_core.services.library import convert_library_concentration_from_ngbyul_to_nm
 
-from ..utils import make_generator, decimal_rounded_to_precision
+from ..utils import decimal_rounded_to_precision
 
-def fetch_sample_data(ids: List[int] =[], queryset=None, query_params=None) -> List:
+class FetchData:
     """
-    Function used to replace the sample serializer across various viewsets.
+    Mixin used to replace the serializer across various viewsets.
+    """
+    fetch_limit = None
+    fetch_offset = None
+
+    def fetch_data(self, ids: List[int] =[]) -> List:
+        """
+        Abstract function to overload to produce serialized data for automated processing.
+
+            Args:
+                self: base_class(viewset) + fetch_data
+                ids: List of ids to select specific objects. Defaults to [].
+
+            Returns:
+                Returns (None, None), overload to get a list and count
+        """
+        # No special filtering for empty ids list otherwise get only the selected ids objects
+        if len(ids) == 1:
+            self.queryset = self.queryset.filter(id=ids[0])
+        elif len(ids) > 1:
+            self.queryset = self.queryset.filter(id__in=ids)
+        # pagination params
+        self.fetch_limit = int(self.request.query_params.get('limit', REST_FRAMEWORK["PAGE_SIZE"]))
+        self.fetch_offset = int(self.request.query_params.get('offset', 0))
+
+        return (None, None) # abstract function, must be overloaded. call base function for initialization
```

Comment:
```suggestion
        return ([], 0) # abstract function, must be overloaded. call base function for initialization
```

Reply:
This would be a valid response ... i want it to crash if someone tries to call the base class and use the return value.